core/src/main/kotlin/org/evomaster/core/languagemodel/LanguageModelConnector.kt

```diff
@@ -0,0 +1,241 @@
+        validatePrompt(prompt)
+
+        val languageModelServerURL = getLanguageModelServerURL()
+
+        val languageModelName = getLanguageModelName()
+
+        val requestBody = objectMapper.writeValueAsString(
+            OllamaRequest(
+                prompt = prompt,
+                stream = false,
+                model = languageModelName.toString()
+            )
+        )
+
+        val response = call(languageModelServerURL, requestBody)
+
+        if (!id.isNullOrBlank()) {
+            if (prompts.contains(id)) {
+                val existingPrompt = prompts[id]
+
+                if (existingPrompt != null) {
+                    existingPrompt?.answer = response
+                }
+            }
+        }
+
+        return response
+    }
+
+    /**
+     * Private method to make the call to the large language model server.
+     *
+     * Note: If you are using Ollama as a server, please make sure to set the
+     * CORS origin for Ollama on the host operating system.
+     *
+     * Reference:
+     * https://medium.com/dcoderai/how-to-handle-cors-settings-in-ollama-a-comprehensive-guide-ee2a5a1beef0
+     */
+    private fun call(languageModelServerURL: String, requestBody: String): String? {
+        try {
+            val connection = URL(languageModelServerURL).openConnection() as HttpURLConnection
+            connection.requestMethod = "POST"
+            // TODO: set to avoid long running calls
+            connection.connectTimeout = 4000
+            connection.setRequestProperty("Content-Type", "application/json")
+            connection.useCaches = false
+            connection.doInput = true
+            connection.doOutput = true
+
+            val writer = DataOutputStream(connection.outputStream)
+            writer.writeBytes(requestBody)
+            writer.flush()
+            writer.close()
+
+            if (connection.responseCode != HttpURLConnection.HTTP_OK) {
+                LoggingUtil.getInfoLogger().warn("Failed to connect to language model server.")
+                return null
+            }
+
+            // This is designed to use the non-stream outputs.
+            // If stream is needed, consider implementing a
+            // different method to handle stream outputs.
+            val response = objectMapper.readValue(connection.inputStream, OllamaResponse::class.java)
+
+            return response.response
+        } catch (e: Exception) {
+            LoggingUtil.getInfoLogger().warn("Failed to connect to language model server: ${e.message}.")
+
+            return null
+        }
+    }
+
+    /**
+     * Private method, returns the large language model server URL from
+     * EMConfig, otherwise returns the possible default local URL.
+     */
+    private fun getLanguageModelServerURL(): String {
+        val languageModelServerURL = if (config.languageModelServerURL == null) {
+            "http://localhost:11434/api/generate"
```

Comment:
is this some test utility? in such case, should not be here. 
otherwise,  what is 11434 representing? is it some defaults of Ollama? clarify